Add SimplenoteIntents extension to app

charliescheer opened the pull request:
### Fix
This PR lays the ground work for adding intent configuration widgets into Simplenote.

As compared to the new note widget, which displays no data, to be able to setup widgets that can be configured by the user we need to create another extension for the app that defines an Intent configuration.  This PR sets that up.

### Test
There isn't much to test with this PR, but you can confirm that the option to configure a widget is there...
There is a placeholder for the note preview widget that was added to be able to confirm the intents were working, so to test this pr, make sure that you have configuration options.  To do that.
1. run the app on device or in the sim to install the widgets
2. On the home screen long press till the home screen editor appears, tap on the plus
3. find simplenote and select the note preview widget
4. Once it the widget is added to your home screen, while home screen edit is enabled, tap on note preview widget.
5. you will see an option to select the note.  Tap on that and you will be given four choices Note 1-4
6. select one and see that the text appears in the widget.

### Review
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
> These changes do not require release notes.

charliescheer committed:
Added basic setup for an intents extension for dynamic widget selection

charliescheer committed:
Added placeholder for note preview widget

charliescheer committed:
Added configuration settings for note widget

charliescheer committed:
Updated version numbers for simplenoteIntents extension

jleandroperez commented on Simplenote.xcodeproj/project.pbxproj:
```diff
@@ -4648,6 +4753,181 @@
+				);
+				MTL_ENABLE_DEBUG_INFO = NO;
+				MTL_FAST_MATH = YES;
+				PRODUCT_BUNDLE_IDENTIFIER = com.codality.NotationalFlow.SimplenoteIntents;
```
We've got:
- **App:** com.codality.NotationalFlow
- **Share Ext:** com.codality.NotationalFlow.Share
- **Widgets:** com.codality.NotationalFlow.Widgets

Perhaps for consistency these should look like `com.codality.NotationalFlow.SimplenoteIntents` ?

jleandroperez commented on Simplenote.xcodeproj/project.pbxproj:
```diff
@@ -4648,6 +4753,181 @@
+			};
+			name = "Distribution Alpha";
+		};
+		BAB5763126703C8200B0C56F /* Distribution AppStore */ = {
```
Another note (mentioned in another PR). We're probably gonna need to poke the Platform team for help, again, with the provisioning profiles here.

jleandroperez commented on Simplenote/Simplenote-Info.plist:
```diff
@@ -74,6 +74,7 @@
 	<string>This app does not require Photo Library access.</string>
 	<key>NSUserActivityTypes</key>
 	<array>
+		<string>SPNoteWidgetIntent</string>
```
Let's drop the `SP` prefix from everywhere :fire:

jleandroperez commented on SimplenoteIntents/Info.plist:
```diff
@@ -0,0 +1,42 @@
+			<array/>
+			<key>IntentsSupported</key>
+			<array>
+				<string>SPNoteWidgetIntent</string>
```
Same note as above, let's drop the `SP` prefix

jleandroperez commented on SimplenoteIntents/IntentHandler.swift:
```diff
@@ -0,0 +1,26 @@
+
+        let placeholderNoteData = ["Note 1", "Note 2", "Note 3", "Note 4"]
+
+        var notes: [SPNote] = []
```
Can't find where `SPNote` is defined, but can we rename that to `IntentNote` or `WidgetNote` or so?

jleandroperez reviewed (commented):
Sending few notes!! First pass looking great!

jleandroperez commented:
Oh! We also need to handle the "Main App is not logged" scenario. No need to address that in this particular PR, just leaving a note, so that we don't loose track of it.

charliescheer committed:
Updated intent bundle identifier to match pattern for other extensions

charliescheer committed:
Added custom class to SPNoteWidgetIntent removing SP prefix

charliescheer committed:
Updated SPNote class to WidgetNote

mokagio committed:
Extract logic to define Simplenote bundle identifiers in Fastlane

mokagio committed:
Add Intents extension bundle identifier to code signing automation

mokagio committed:
Explicitly set provisioning profile for Intents extension in Debug

See
https://github.com/Automattic/simplenote-ios/pull/1344#pullrequestreview-699434538

mokagio committed:
Use Manual code signing for Intents extension

This also required explicitly specifying the provisioning profiles for
each build configuration.

mokagio committed:
DRY simplenote_app_identifiers implementation in Fastfile

Hat tip @AliSoftware.

See
https://github.com/Automattic/simplenote-ios/pull/1344#discussion_r666023124

mokagio committed:
DRY Fastlane logic to define the provisioning profiles mapping

mokagio committed:
Move free floating constant together with others in Fastfile

Thanks for the suggestion @AliSoftware

See
https://github.com/Automattic/simplenote-ios/pull/1344#discussion_r666067016

mokagio committed:
Gold shaving: Replace array assignment with array transform in Fastfile

mokagio committed:
Drop a redundant parameter usage in Fastfile

The value was the same as the default, and in another part of the code
for a different method we do not pass the parameter when it's equal to
the default.

charliescheer committed:
Merge pull request #1344 from Automattic/feature/widget-intents-code-signing-automation

Update Fastlane code signing automation to include Intents extension

charliescheer commented:
Most of the work on this PR was done in #1344  merging this back into the feature branch

jleandroperez commented on SimplenoteWidgets/Providers/NotePreviewWidgetProvider.swift:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    func getSnapshot(for configuration: NoteWidgetIntent, in context: Context, completion: @escaping (NotePreviewWidgetEntry) -> Void) {
+        let entry = NotePreviewWidgetEntry(date: Date(), text: "Placeholder")
```
Not locallized string

jleandroperez commented on SimplenoteWidgets/Providers/NotePreviewWidgetProvider.swift:
```diff
@@ -0,0 +1,32 @@
+    typealias Entry = NotePreviewWidgetEntry
+
+    func placeholder(in context: Context) -> NotePreviewWidgetEntry {
+        return NotePreviewWidgetEntry(date: Date(), text: "Placeholder")
```
Not localized string

jleandroperez reviewed (commented):
Hey there @charliescheer !!

Found a couple strings that aren't localized (and I belive those aren't SwiftUI code)
